src/UbidotsEthernet.cpp

```diff
@@ -0,0 +1,349 @@
+      }
+      _client.stop();
+      free(data);
+      return ERROR_VALUE;
+    }
+    res[i++] = c;
+  }
+
+  /* Parses the response to get just the last value received */
+  response = strtok(res, "\r\n");
+  while(response!=NULL) {
+    j++;
+    printf("%s", response);
+    response = strtok(NULL, "\r\n");
+    /* Saves the last value obtained in a new variable */
+    if (j == 10) {
+      if (response != NULL) {
+        serverResponse = response;
+      }
+      j = 0;
+    }
+  }
+  /* Converts the value obtained to a float */
+  num = atof(serverResponse);
+  /* Free memory*/
+  free(data);
+  free(res);
+  /* Removes any buffered incoming serial data */
+  _client.flush();
+  /* Disconnects the client */
+  _client.stop();
+  return num;
+}
+
+/***************************************************************************
+FUNCTIONS TO SEND DATA
+***************************************************************************/
+
+ /**
+  * Add a value of variable to save
+  * @arg variable_label variable label to save in a struct
+  * @arg value variable value to save in a struct
+  * @arg ctext [optional] is the context that you will save, default
+  * @arg timestamp_val [optional] is the timestamp for the actual value
+  * is NULL
+  */
+void Ubidots::add(char *variable_label, double value) {
+  return add(variable_label, value, NULL, NULL);
+}
+
+void Ubidots::add(char *variable_label, double value, char *ctext) {
+  return add(variable_label, value, ctext, NULL);
+}
+
+void Ubidots::add(char *variable_label, double value, char* ctext, long unsigned timestamp_val ) {
+  (val+currentValue)->varLabel = variable_label;
+  (val+currentValue)->varValue = value;
+  (val+currentValue)->context = ctext;
+  (val+currentValue)->timestamp_val = timestamp_val;
+  currentValue++;
+  if(currentValue>maxValues) {
+    Serial.println(F("You are sending more than 5 consecutives variables, you just could send 5 variables. Then other variables will be deleted!"));
+    currentValue = maxValues;
+  }
+}
+
+/**
+ * Assigns a new device label
+ * @arg new_device_name new label that you want to assign to your device
+ */
+void Ubidots::setDeviceLabel(char* new_device_label) {
+    _deviceLabel = new_device_label;
+}
+
+/**
+ * Gets the length of the body
+ * @arg body data to be send - JSON
+ * @return dataLen the length of the JSON
+ */
+int Ubidots::dataLen(char* body) {
+  uint8_t dataLen = 0;
+  for (int i = 0; i <= 150; i++) {
+    if (body[i] != '\0') {
+      dataLen++;
+    } else {
+      break;
+    }
+  }
+  return dataLen;
+}
+
+/**
+ * Send all data of all variables that you saved
+ * @reutrn true upon success, false upon error.
+ */
+bool Ubidots::sendAll() {
+  /* Assigns the constans as global on the function */
+  uint8_t max_retries = 0;
+  uint8_t timeout = 0;
+  uint8_t i = 0;
+  String str;
+
+  /* Saves variable value in str */
+  str = String(((val+i)->varValue),3); // variable value
```

Comment:
@mariacarlinahernandez posible overflow, what happen when `val` has no values and the user invoke sendAll?